pennylane/devices/default_qubit.py

```diff
@@ -451,9 +459,13 @@ def preprocess(
         transform_program = TransformProgram()
 
         transform_program.add_transform(validate_device_wires, self.wires, name=self.name)
-        transform_program.add_transform(qml.defer_measurements, device=self)
+        transform_program.add_transform(mid_circuit_measurements, device=self)
+        stop_cond = stopping_condition if not self.shots else stopping_condition_shots
```

Comment:
This is something I am we should not do, as shots are purely determined by the tapes themselves and can be set independently of `Device.shots`. The point of `Device.shots` is to initialize the start of the workflow, not actually determine the number of shots that will be used.